sunpy/map/sources/tests/test_eui_source.py

```diff
@@ -0,0 +1,265 @@
+        NAXIS1  =                  768
+        NAXIS2  =                  768
+        LONGSTRN= 'OGIP 1.0'           / The OGIP long string convention may be used
+        FILENAME= 'solo_L1_eui-fsi304-image_20201021T145510206_V01.fits' / FITS filename
+        DATE    = '2020-11-23T16:40:54.714' / [UTC] FITS file creation date
+        FILE_RAW= 'BatchRequest.PktTmRaw.SOL.0.2020.295.15.15.01.857.eJeU@2020.295.15.&'
+        CONTINUE  '15.03.463.1.xml&'
+        CONTINUE  '' / raw filename
+        PARENT  = 'solo_L0_eui-fsi###-image_0656607273e84f_V00.fits' / source file curre
+        APID    =                  924 / APID number of associated TM
+        DATE-OBS= '2020-10-21T14:55:10.206' / [UTC] deprecated, same as DATE-BEG.
+        DATE-BEG= '2020-10-21T14:55:10.206' / [UTC] start time of observation
+        DATE-AVG= '2020-10-21T14:55:13.206' / [UTC] average time of observation
+        TIMESYS = 'UTC     '           / system used for time keywords
+        OBT_BEG =    656607273.9074554 / start acquisition time in OBT
+        LEVEL   = 'L1      '           / data processing level
+        ORIGIN  = 'Royal Observatory of Belgium' / file generation location
+        VERS_SW = '807     '           / version of SW that provided FITS file
+        VERSION = '01      '           / incremental version number
+        IMGTYPE = 'solar image'        / type of image; solar, calib., engineering
+        JOBID   = '20201123T163957.998Z_01f' / unique pipeline job ID
+        COMPLETE= 'C       '           / C or I for complete/incomplete
+        OBSRVTRY= 'Solar Orbiter'      / satellite name
+        TELESCOP= 'SOLO/EUI/FSI'       / telescope/Sensor name
+        INSTRUME= 'EUI     '           / instrument name
+        DETECTOR= 'FSI     '           / instrument subunit or sensor
+        DOORINT = 'open    '           / internal door position
+        DOOREXT = 'open    '           / external HS Door 2 FSI @-76.08 s
+        XPOSURE =                  6.0 / [s] total effective exposure time
+        FILTER  = 'Magnesium_304_2'    / filter position
+        WAVELNTH=                  304 / [Angstrom] characteristic wavelength observatio
+        WAVEMIN =                  250 / [Angstrom] min wavelength resp. > 0.05 of max
+        WAVEMAX =                  350 / [Angstrom] max wavelength resp. > 0.05 of max
+        SOOPNAME= 'not defined'        / name of the SOOP Campaign that the data belong
+        SOOPTYPE= '000     '           / campaign ID(s) that the data belong to
+        OBS_MODE= 'GENERIC_CALIB'      / observation mode
+        OBS_TYPE= '2ZpG    '           / encoded version of OBS_MODE
+        OBS_ID  = 'SEUI_021A_000_000_2ZpG_11K' / unique ID of the individual observation
+        TARGET  = 'not defined'        / type of target from planning
+        BTYPE   = 'Intensity'          / type of data
+        BUNIT   = 'ct      '           / units of physical value, after BSCALE, BZERO
```

Comment:
Did you fix this header? or if not why is there the unit fix in the constructor?

Reply:
I cheated and roundtripped the file through sunpy first to fix the `WAVEUNIT` issue.